base/applications/network/dwnl/dwnl.c

```diff
@@ -246,6 +246,14 @@ CBindStatusCallback_OnProgress(IBindStatusCallback *iface,
             break;
 
         case BINDSTATUS_ENDDOWNLOADDATA:
+            /* since download is completed, update progress one last time to be at 100% */
```

Comment:
```suggestion
            /* Since download is completed, update progress one last time to be at 100% */
```

Reply:
I also wonder whether this couldn't be simplified to just:
```c
    This->Progress = This->Size;
    CBindStatusCallback_UpdateProgress(This); // Show 100% progress now
```
without any `This->Size` check ?
The reasons:
- simplifies this part of the code;
- `CBindStatusCallback_UpdateProgress` already checks whether `This->Size` is zero or not
- and if Size is zero, `CBindStatusCallback_UpdateProgress` could still be called and [its `Size == 0` branch would be taken](https://github.com/reactos/reactos/pull/7903/files#diff-927900822615b14feda0a6e0edd0a6e15dad3cc08735458e444087d2364014d0R83), thus showing the very last (final) file size that has been downloaded.

Reply:
* Suggested change 1 was made: c0c2dd2cd03188e681fd9e0f81bdb17917a4fdfb
* Suggested change 2 (remove check): Yes, thank you, you are right! I guess I was more on the safer side. Anyway, addressed: [dwnl-progress-fix-main 9a2b87733e0] Remove redundant check